packages/pages/src/dev/server/middleware/serverRenderRoute.ts

```diff
@@ -57,15 +57,16 @@ export const serverRenderRoute =
           featuresConfig,
           dynamicGenerateData,
           projectStructure,
-        }
-      );
+        });
 
       // render the component to its html
       // Since we are on the server using plain TS, and outside
       // of Vite, we are not using JSX here
-      const appHtml = await ReactDOMServer.renderToString(
-        React.createElement(Component, props)
-      );
+      const appHtml = render
+        ? render(props)
```

Comment:
similar comment to entry but the expected return of the render function is the whole file -- not just the html body like the default export